### PR TITLE
feat: adds mainnet watcher and executor script

### PR DESCRIPTION
**Pull request by b0dhidharma**

### CODE REVIEW

File: packages/yswaps/scripts/execute-mainnet-pending-trades.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    let bestSetup: TradeSetup = tradesSetup[0];
+
+    for (let i = 1; i < tradesSetup.length; i++) {
+      if (tradesSetup[i].minAmountOut!.gt(bestSetup.minAmountOut!)) {
+        bestSetup = tradesSetup[i];
+      }
+    }
+
+    const tx = await tradeFactory
+      .connect(flashBotSigner)
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/execute-mainnet-pending-trades.ts:94]: @lbertenasco have you ever tested this approach ?

Comment (lbertenasco) [reply packages/yswaps/scripts/execute-mainnet-pending-trades.ts:94]: haven't tested it, but you should use block protection contract.
https://etherscan.io/address/0xcc268041259904bb6ae2c84f9db2d976bceb43e5

Comment (lbertenasco) [reply packages/yswaps/scripts/execute-mainnet-pending-trades.ts:94]: you'll have more control by manually submitting to flashbots, rpc is a bit buggy specially on tx replacement.

**Review by lbertenasco — CHANGES_REQUESTED** [review]
Use manual flashbot bundle submition.
Potentially add on-chain uncle block protection

**Comment by b0dhidharma** [comment]
@lbertenasco why not use the RPC ? Flashbot bundle submittion is a lot of overhead if RPC provides it out of the box. Am I missing something ?

**Comment by lbertenasco** [comment]
> @lbertenasco why not use the RPC ? Flashbot bundle submittion is a lot of overhead if RPC provides it out of the box. Am I missing something ?

@b0dhidharma transaction replacement is a bit buggy, we should do manual bundle creation and submition to the API directly.
a good utils library to handle this would be a really nice touch

**Pull request merged**